apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/ConfigsImportService.java

```diff
@@ -0,0 +1,139 @@
+      return true;
+    } else {
+      return permissionValidator.hasModifyNamespacePermission(appId, namespace, env);
+    }
+  }
+
+  /**
+   * move from {@link com.ctrip.framework.apollo.portal.controller.ConfigsImportController}
+   */
+  public void importConfig(
+      final String appId,
+      final String env,
+      final String clusterName,
+      final String namespaceName,
+      final long namespaceId,
+      final String format,
+      final String configText
+  ) {
+    final NamespaceTextModel model = new NamespaceTextModel();
+
+    model.setAppId(appId);
+    model.setEnv(env);
+    model.setClusterName(clusterName);
+    model.setNamespaceName(namespaceName);
+    model.setNamespaceId(namespaceId);
+    model.setFormat(format);
+    model.setConfigText(configText);
+
+    itemService.updateConfigItemByText(model);
+  }
+
+  /**
+   * import one config from file
+   */
+  public void importOneConfigFromFile(
+      final String appId,
+      final String env,
+      final String clusterName,
+      final String namespaceName,
+      final String configText,
+      final String format
+  ) {
+    final NamespaceDTO namespaceDTO = namespaceService
+        .loadNamespaceBaseInfo(appId, Env.valueOf(env), clusterName, namespaceName);
+    this.importConfig(appId, env, clusterName, namespaceName, namespaceDTO.getId(), format, configText);
+  }
+
+  /**
+   * import a config file.
+   * the name of config file must be special like
+   * appId+cluster+namespace.format
+   * Example:
+   * <pre>
+   *   123456+default+application.properties (appId is 123456, cluster is default, namespace is application, format is properties)
+   *   654321+north+password.yml (appId is 654321, cluster is north, namespace is password, format is yml)
+   * </pre>
+   * so we can get the information of appId, cluster, namespace, format from the file name.
+   * @param env environment
+   * @param standardFilename appId+cluster+namespace.format
+   * @param configText config content
+   */
+  public void importOneConfigFromText(
+      final String env,
+      final String standardFilename,
+      final String configText
+  ) {
+    final String appId = ConfigFileUtils.getAppId(standardFilename);
+    final String clusterName = ConfigFileUtils.getClusterName(standardFilename);
+    final String namespace = ConfigFileUtils.getNamespace(standardFilename);
+    final String format = ConfigFileUtils.getFormat(standardFilename);
+    this.importOneConfigFromFile(appId, env, clusterName, namespace, configText, format);
+  }
+
+  /**
+   * @see ConfigsImportService#importOneConfigFromText(java.lang.String, java.lang.String, java.lang.String)
+   * @throws AccessControlException if has no modify namespace permission
+   */
+  public void importOneConfigFromFile(
+      final String env,
+      final String standardFilename,
+      final InputStream inputStream
+  ) {
+    // check permission
+    if (!this.hasModifyNamespacePermission(env, standardFilename)) {
```

Comment:
This is not necessary because the permission is first checked in com.ctrip.framework.apollo.portal.controller.ConfigsImportController#importConfigFile